cli/release.go

```diff
@@ -149,3 +156,24 @@ func runReleaseAddDocker(args *docopt.Args, client *controller.Client) error {
 
 	return nil
 }
+
+func canonicalizeDockerURL(rawurl string) (string, error) {
+	u, err := url.Parse(rawurl)
+	if err != nil {
+		return "", err
+	}
+	q := u.Query()
+	if tag := q.Get("tag"); tag != "" {
+		if ref, err := registry.NewRef(rawurl); err != nil {
+			return "", err
+		} else if image, err := registry.NewDockerSession(ref).GetImage(); err != nil {
```

Comment:
We may as well fix #1568 by always doing this (rather than just when `tag` is set).

Given this takes a few HTTP requests to complete, it would be useful to print a message like `validating Docker image URI`, so the output is something like:

```
$ flynn release add https://registry.hub.docker.com?name=ubuntu&tag=foo
validating Docker image URI
ERROR: unknown tag "foo"

$ flynn release add https://registry.hub.docker.com?name=bar&tag=foo
validating Docker image URI
ERROR: unknown image "bar"

$ flynn release add https://example.com?name=bar&tag=foo
validating Docker image URI
ERROR: invalid registry "example.com"
```